frontend/ui/widget/imageviewer.lua

```diff
@@ -647,13 +647,16 @@ function ImageViewer:onZoomIn(inc)
         inc = 0.2
     end
 
+    -- Compute new scale factor for rescaled image dimensions
+    local new_factor = self._image_wg:recomputeScaleFactor(1 + inc)
+
     -- We destroy ImageWidget on update, so only request this the first time,
     -- in order to avoid jitter in the results given differing memory consumption at different zoom levels...
     if not self._max_scale_factor then
         self._min_scale_factor, self._max_scale_factor = self._image_wg:getScaleFactorExtrema()
     end
     -- Clamp to sane values
-    local new_factor = math.min(self.scale_factor + inc, self._max_scale_factor)
+    new_factor = math.min(self.scale_factor + inc, self._max_scale_factor)
```

Comment:
You're not using new_factor here (it's correct I think it the onZoomOut section).

Reply:
Yeah, see latest commit ;).